### PR TITLE
Implement `TrustedLen` for `Fuse<I: TrustedLen>`

### PR DESCRIPTION
**Pull request by sdroege**
This looks like it was simply forgotten.

### CODE REVIEW

File: library/core/src/iter/adapters/fuse.rs [hunk 1]
```diff
@@ -182,6 +184,9 @@ where
     }
 }
 
+#[unstable(feature = "trusted_len", issue = "37572")]
+unsafe impl<I> TrustedLen for Fuse<I> where I: TrustedLen {}
```
Comment (Aaron1011) [inline library/core/src/iter/adapters/fuse.rs:188]: Could you add a `// SAFETY:` comment?

Comment (sdroege) [reply library/core/src/iter/adapters/fuse.rs:188]: Indeed, also for the `TrustedRandomAccess` below. That `TrustedLen` is OK is a weaker requirement than `TrustedRandomAccess`

Comment (sdroege) [reply library/core/src/iter/adapters/fuse.rs:188]: Done

**Pull request merged**